src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.macios.cs

```diff
@@ -99,40 +99,41 @@ public static void SetLayout(this MauiPopup mauiPopup, in IPopup popup)
 
 		if (popup.Anchor is null)
 		{
-			var originY = popup.VerticalOptions switch
+			nfloat originY;
+			if (mauiPopup.PreferredContentSize.Height < frame.Height)
 			{
-				Microsoft.Maui.Primitives.LayoutAlignment.End => frame.Height,
-				Microsoft.Maui.Primitives.LayoutAlignment.Center => frame.GetMidY(),
-				_ => 0f
-			};
-
-			var originX = popup.HorizontalOptions switch
+				originY = popup.VerticalOptions switch
+				{
+					Microsoft.Maui.Primitives.LayoutAlignment.Start => mauiPopup.PreferredContentSize.Height / 2,
+					Microsoft.Maui.Primitives.LayoutAlignment.End => frame.Height - (mauiPopup.PreferredContentSize.Height / 2),
+					Microsoft.Maui.Primitives.LayoutAlignment.Center => frame.GetMidY(),
+					_ => frame.GetMidY()
+				};
+			}
+			else
 			{
-				Microsoft.Maui.Primitives.LayoutAlignment.End => frame.Width,
-				Microsoft.Maui.Primitives.LayoutAlignment.Center => frame.GetMidX(),
-				_ => 0f
-			};
+				originY = -frame.GetMidY();
+			}
 
-			if (DeviceInfo.Current.Platform == DevicePlatform.MacCatalyst)
+			nfloat originX;
+			if (mauiPopup.PreferredContentSize.Width < frame.Width)
 			{
-				if (popup.VerticalOptions == Microsoft.Maui.Primitives.LayoutAlignment.End)
-				{
-					originY -= (mauiPopup.PreferredContentSize.Height / 2);
-				}
-
-				if (popup.HorizontalOptions == Microsoft.Maui.Primitives.LayoutAlignment.End)
-				{
-					originX -= (mauiPopup.PreferredContentSize.Width);
-				}
-
-				if (popup.HorizontalOptions == Microsoft.Maui.Primitives.LayoutAlignment.Center)
+				originX = popup.HorizontalOptions switch
 				{
-					originX -= (mauiPopup.PreferredContentSize.Width / 2);
-				}
+					Microsoft.Maui.Primitives.LayoutAlignment.Start => mauiPopup.PreferredContentSize.Width / 2,
+					Microsoft.Maui.Primitives.LayoutAlignment.End => frame.Width - (mauiPopup.PreferredContentSize.Width / 2),
+					Microsoft.Maui.Primitives.LayoutAlignment.Center => frame.GetMidX(),
+					_ => frame.GetMidX()
+				};
+			}
+			else
+			{
+				originX = -frame.GetMidX();
 			}
 
 			mauiPopup.PopoverPresentationController.SourceRect = new CGRect(originX, originY, 0, 0);
 			mauiPopup.PopoverPresentationController.PermittedArrowDirections = 0;
+			mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(1, 0, 0, 0);
```

Comment:
Just curious - how come we are now adding 1 point of padding to the top `PopoverLayoutMargins`?

Reply:
If you set everything to 0 including the Top of PopupLayoutMargins, isn't it because the margin to close when you tap outside the Popup area is completely gone?
I think that's probably the reason. By the way, there is no problem even if Top is 0.1 instead of 1.

It seems that setting everything to 0 and eliminating the margin itself is not allowed. By the way, if you set everything including PopoverLayoutMargins to 0, unintended margins will be taken.

This is the execution result when all PopoverLayoutMargins values ​​are set to 0.
![2023-08-07_05-07-34-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/9b264cb9-059c-424c-92e4-8e9e84cc7e02)


I'll leave it up to you to decide what to do.

Reply:
Hi @cat0363! I'm not sure I understand the problem. 

Could you provide an example or point to an GitHub Issue to help me understand? 

I'm ready to merge this PR once we can resolve this conversation.

Reply:
Hi, @brminnick 

If you specify 0 for the top, bottom, left, and right of PopoverLayoutMargins, margins are added.
A fixed size margin is always added.

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0, 0, 0, 0);

![2023-08-08_02-30-19-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/e71bdfb3-7b02-4b38-9ea7-dc7dda906539)

Originally, I want the top, bottom, left, and right margins to be 0 as shown below.

![2023-08-08_03-37-34-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/cb1e3147-5173-4b93-a22c-da354f3457a2)

But it doesn't work like above.
This is probably because if you set the top, bottom, left, and right margins to 0 as described above, you will not be able to close the Popup by tapping outside it.

It seems that the top, bottom, left, and right margins are intentionally added so as not to impair operability.
Specifying all 0 for the top, bottom, left, and right of the PopoverLayoutMargins intentionally adds a margin, so you need to add a small margin somewhere on the top, bottom, left, and right.

We need an area to tap outside the popup to close it, so we need to reserve that area.
All of the following have small margins on the top, bottom, left, and right, but not large margins like when all 0 is specified.

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(1, 0, 0, 0);
    
![2023-08-08_03-58-55-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/d13fa33c-98e6-476b-9b54-f34468077cfb)

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0, 1, 0, 0);

![2023-08-08_04-01-40-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/fed7db2b-6570-4cbb-bcfb-927607cc3998)

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0, 0, 1, 0);

![2023-08-08_04-03-28-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/ec40acbc-d8bf-4d42-acd5-f8ee33cf8830)

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0, 0, 0, 1);

![2023-08-08_04-08-40-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/68b05082-56c0-44f0-b2c9-0d23bfc2d0ad)

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(1, 1, 1, 1);

![2023-08-08_04-10-54-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/f5ecdd30-3198-4ff6-9d00-a664ccbdec5a)

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0.1f, 0.1f, 0.1f, 0.1f);

![2023-08-08_04-13-57-午後](https://github.com/CommunityToolkit/Maui/assets/125236133/5f733579-1f2b-4595-8e09-5e274616cf9e)

You don't have to specify a value other than 0 only for Top.
The reason is that it is sufficient if there is an area that can be tapped somewhere on the top, bottom, left, or right.

If you want to make it look like there is no margin, it might be better to specify a value evenly close to 0.

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0.1f, 0.1f, 0.1f, 0.1f);

However, if it is as close to 0 as possible, it will be difficult to tap the outside of the Popup to close it, so be careful. Therefore, it is necessary to consider how much the margin should be.

It is unclear if the margin added by default is 10 points as described below.

https://developer.apple.com/documentation/uikit/uipopoverpresentationcontroller/1622323-popoverlayoutmargins

I added the following project to the modified CommunityToolkit project, changed the value of PopoverLayoutMargins, and verified the operation.

https://github.com/cat0363/MauiComm-IssuePopupSizePosition.git

Reply:
If there is a margin, it is easy to close the Popup as shown in the video below, but if there is no margin, it will be difficult to close the Popup.

https://github.com/CommunityToolkit/Maui/assets/125236133/cf082a01-a382-4fef-a6d8-fb765ea393bd

The following is a video when the margin is infinitely close to 0.

https://github.com/CommunityToolkit/Maui/assets/125236133/56f7e32d-f463-4b24-ac4d-18a70720b1c8

It was possible to close only when tapping the lower safe area part of the top, bottom, left, and right.
By the way, on devices like the iPhone SE, the outer area of ​​the Popup was too small to close.

Care should be taken if the margin is too close to 0. Because there is no back button in iOS like in Android.
We should consider whether or not you should set a certain margin.

Reply:
Thanks @cat0363. 

If I understand correctly, your concern is that when devs create a full-screen Popup that their user won't be able to tap outside of the popup to dismiss it. 

I disagree that we should dictate this behavior in the library; it is the developer's prerogative. In other words, if a developer creates a full-screen Popup because they don't want their user to tap outside of it, we shouldn't prevent that. The developer can add a button to the popup, like we demonstrate in [`ButtonPopup` in `CommunityToolkit.Maui.Samples`](https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Views/Popups/ButtonPopup.xaml), to dismiss the Popup for that use-case.

We are planning the release of `CommunityToolkit.Maui v5.3.0` in a few hours and I'd really like to include this PR in the release. I'm going to remove the below line of code from this PR to meet our release deadline, however I'm more than happy to be proven wrong about this. Feel free to open a Discussion or an Issue where we can dive deeper into the problem and where other folks from the community can collaborate with us.

```cs
mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(1, 0, 0, 0);
```

Reply:
Hi, @brminnick 
I agree with you about leaving the popup closing method up to the developer.
If an issue is raised regarding the Margin of Popup after the release, in that case, try setting values ​​close 
to 0 for the top, bottom, left, and right of the PopoverLayoutMargins. For example:

    mauiPopup.PopoverPresentationController.PopoverLayoutMargins = new UIEdgeInsets(0.1f, 0.1f, 0.1f, 0.1f);

If you remove the PopoverLayoutMargins code, the default margins on the iOS side will be added. 
So the Popup doesn't cover the entire screen. However, if you don't want to close the Popup by 
tapping the margin, you can control this using the CanBeDismissedByTappingOutsideOfPopup property. 

I look forward to the release of the latest version of the Community Toolkit.